tests/gdn/test_decode_ucache.py

```
@@ -0,0 +1,499 @@
+    # commit mode with a non-contiguous base: was a lost commit inside a
+    # silent .contiguous() copy, now an assertion.
+    cb_col = torch.zeros(B, 2, dtype=torch.int32, device=DEV)[:, 0]
+    assert not cb_col.is_contiguous()
+    with pytest.raises(AssertionError, match="contiguous"):
+        mod.gated_delta_rule_mtp_ucache_flush(
+            A_log, a, dt_bias, hist_len=hl.clone(), cache_base=cb_col,
+            restart_hist_on_flush=True, **common)
+
+    # oversized window: was silent wrong output + corrupted fold (17..28)
+    # or a revived append-into-window race (>= 29), now an assertion.
+    hl_bad = hl.clone(); hl_bad[1] = 17
```

Ruff (0.15.21)

[error] Multiple statements on one line (semicolon)

(E702)